### PR TITLE
Fix/transfer

### LINKED ISSUE #8739: Transfer not working
this guy 

telegram-cloud-photo-size-1-4911512188259249079-y
telegram-cloud-photo-size-1-4911512188259249080-y
telegram-cloud-photo-size-1-4911512188259249081-y
after reload
telegram-cloud-photo-size-1-4911469384615177247-y
- it burns the fee but the transfer fails 

link to message 
https://t.me/kodadot_eco/3975/8312

its on polkadot relay chain

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [x] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #8739

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

## Screenshot 📸

- [ ] My fix has changed UI

### CODE REVIEW

**Comment by daiagi** [comment]
Main logic addition is taking into account transaction fees and ED when calculating the transferable balance

as for refactoring, 2 things:
1. uniting various computed refs that were handling token/usd display, with or without fees etc,  into easier to handle objects, so we can use clear syntax such as `total.withoutFee.usd`
3. just moving things around, putting refs on top, then computed refs and all the watchers at the bottom, no logic changed

File: components/transfer/Transfer.vue [hunk 1]
```diff
@@ -379,46 +379,160 @@ const { fetchFiatPrice, getCurrentTokenValue } = useFiatStore()
...
 
-watch(isLoading, (newValue, oldValue) => {
-  // trigger modal only when loading change from false => true
-  // we want to keep modal open when loading changes true => false
-  if (newValue && !oldValue) {
-    isLoaderModalVisible.value = isLoading.value
-  }
-})
+const DOT_BUFFER_FEE = 10000000 // 0.001
+const KSM_BUFFER_FEE = 100000000 // 0.0001
```
Comment (roiLeo) [inline components/transfer/Transfer.vue:387]: **suggestion**: move to `libs/static`

**Review by roiLeo — APPROVED** [review]
isn't `getActionTransactionFee()` what we are looking for?

File: components/transfer/Transfer.vue (hunk 1 above)
Comment (daiagi) [reply components/transfer/Transfer.vue:387]: I'll do it in a follow up after this and #8740 are merged
don't want to create merge conflicts

**Comment by daiagi** [comment]
> isn't getActionTransactionFee() what we are looking for?

maybe, but the way we get the txFee is besides the point in this PR and i didn't change that.
it was more about taking ED and fee into account, so we prevent situation where after paying fee there isn't enough balance to cover the transfer

or am i missing your point?

**Comment by prury** [comment]
- User should not be able to transfer amounts beyond ED anymore? (only 0.36 transferable)

image

**Comment by daiagi** [comment]
@prury

i would suggest a follow up for that to create interface like talisman
where you have a max button that triggers a `transfer_all` extrinsic, assuring no loss on sending account

we have something similar on teleport page

but, i don't know if it is a high prio, since we are an nft marketplace, not a wallet, and talisman wallet (and others i guess) already  is doing this work very well


so in the meantime, at least user is prevented from doing mistakes that will cause money lost

**Comment by prury** [comment]
> @prury
> 
> i would suggest a follow up for that to create interface like talisman where you have a max button that triggers a `transfer_all` extrinsic, assuring no loss on sending account
> 
> what's nice about `transfer_all` is that , afaik, it will deduct the fee from the amount sent, so we don't have to deal with it
> 
> we have something similar on teleport page (not using `transfer_all` to be clear)
> 
> but, i don't know if it is a high prio, since we are an nft marketplace, not a wallet, and talisman wallet (and others i guess) already is doing this work very well
> 
> so in the meantime, at least user is prevented from doing mistakes that will cause money lost

i agree, will create low priority issue for it

**Comment by prury** [comment]
Dot: 
https://polkadot.subscan.io/extrinsic/0xa9e3477ab8a1622bc486f1be8fb9d44bde88677ef5fdb6aceea16bf86566e8e4 ✔️ 
regarding network fee, it worked well, true transaction fee was $0.14 but UI showed me a bit more, like $0.20

Kusama
https://kusama.subscan.io/tx/0xa4692d20cca48f3e728e6e72bebbee812836eb1128d5ca8e6b8ba78c1bc87b6a ✔️

**Comment by yangwao** [comment]
Thanks!
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 8.37 USD/DOT ~ 5.974 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0x54d6ec479bf423c4b183a4bc1aacc00362cefde322091d5226734a05f9730005](https://polkadot.subscan.io/extrinsic/0x54d6ec479bf423c4b183a4bc1aacc00362cefde322091d5226734a05f9730005)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**